### PR TITLE
change: Add label_headers option for Clarify ModelExplainabilityMonitor

### PR DESCRIPTION
**Pull request by xgchena**
*Description of changes:*

For Clarify explainability job, the label_headers option is used to beautify the analysis report by replacing placeholders like "label0". The option has been added to SageMakerClarifyProcessor API by PR https://github.com/aws/sagemaker-python-sdk/pull/2446, this commit adds the same option to ModelExplainabilityMonitor.

*Testing done:*

* Unit Test passed
* Integ Test passed and the label name in analysis reports is replaced as expected.

For tracking: https://tiny.amazon.com/1i1p4eh9w

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [X] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [X] I certify that the changes I am introducing will be backword compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [X] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [X] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [X] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [X] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [X] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [X] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [X] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by larroy — DISMISSED** [review]
minor comment, LGTM

**Comment by xgchena** [comment]
Rebased to resolve conflicts.

**Comment by xgchena** [comment]
> ### AWS CodeBuild CI Report
> * CodeBuild project: sagemaker-python-sdk-slow-tests
> * Commit ID: [dd954b1](https://github.com/aws/sagemaker-python-sdk/commit/dd954b1e310e28cb93a90300d780d5348c95419b)
> * Result: FAILED
> * [Build Logs](https://we1smikb89.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=ee33058a-3d2f-47f5-b57f-af9280aae184%2Fbuild.log) (available for 30 days)
> 
> _Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)_

The log is incomplete/not showing the failure.

**Comment by xgchena** [comment]
No change, force push to rebuild and hope this time the slow tests either pass or show full build log.

**Comment by xgchena** [comment]
Rebase and force push to trigger checks.

**Comment by xgchena** [comment]
Changed target branch to dev which is now the default branch.
Rebased from latest dev branch code.
The change was reviewed and approved by Pedro, after that there are only rebases and minor changes to resolve conflicts.

**Comment by xgchena** [comment]
Shreya granted me access to the CodeBuild account to view build logs.

Below are the failure reasons and I can confirm they are not caused by changes of this PR.

* AWS CodeBuild us-west-2 (sagemaker-python-sdk-notebook-tests)

  One notebook failed,

```
--------------------------------------------------------------------------------
SUCCEEDED   11
FAILED  1
ERROR   0
SKIPPED 0
CANCELLED   0
TIMED_OUT   0
--------------------------------------------------------------------------------
...
FAILED  ./amazon-sagemaker-examples/sagemaker-pipelines/tabular/abalone_build_train_deploy/sagemaker-pipelines-preprocess-train-evaluate-batch-transform.ipynb  PT31M34.794S
...
```

* AWS CodeBuild us-west-2 (sagemaker-python-sdk-pr)

  Four test cases failed,

```
=========================== short test summary info ============================
FAILED tests/integ/test_kmeans.py::test_kmeans - botocore.exceptions.ClientEr...
FAILED tests/integ/test_multidatamodel.py::test_multi_data_model_deploy_pretrained_models
FAILED tests/integ/test_multidatamodel.py::test_multi_data_model_deploy_pretrained_models_update_endpoint
FAILED tests/integ/test_multidatamodel.py::test_multi_data_model_deploy_trained_model_from_framework_estimator[1.8.0-py37]
= 4 failed, 282 passed, 24 skipped, 23448 warnings, 82 rerun in 5607.57s (1:33:27) =
```

* AWS CodeBuild us-west-2 (sagemaker-python-sdk-slow-tests)

  This test failed due to "BUILD_TIMED_OUT: Build has timed out. "

File: src/sagemaker/clarify.py [hunk 1]
```diff
@@ -1057,13 +1061,12 @@ def run_explainability(
...
             model_config (:class:`~sagemaker.clarify.ModelConfig`): Config of the model and its
                 endpoint to be created.
-            explainability_config (:class:`~sagemaker.clarify.ExplainabilityConfig` or list):
-                Config of the specific explainability method or a list of ExplainabilityConfig
-                objects. Currently, SHAP and PDP are the two methods supported.
-            model_scores(str|int|ModelPredictedLabelConfig):  Index or JSONPath location in the
-                model output for the predicted scores to be explained. This is not required if the
-                model output is a single score. Alternatively, an instance of
-                ModelPredictedLabelConfig can be provided.
+            explainability_config (:class:`~sagemaker.clarify.ExplainabilityConfig`): Config of the
```
Comment (navinsoni) [inline src/sagemaker/clarify.py:1064]: Is this backward compatible. List is removed and it do not support PDP anymore

File: src/sagemaker/clarify.py [hunk 2]
```diff
@@ -1057,13 +1061,12 @@ def run_explainability(
...
+                specific explainability method. Currently, only SHAP is supported.
+            model_scores (int or str or :class:`~sagemaker.clarify.ModelPredictedLabelConfig`):
+                Index or JSONPath to locate the predicted scores in the model output. This is not
+                required if the model output isa single score. Alternatively, it can be an instance
```
Comment (navinsoni) [inline src/sagemaker/clarify.py:1068]: Please give space between "is a"

File: src/sagemaker/model_monitor/clarify_model_monitoring.py [hunk 3]
```diff
@@ -833,9 +833,10 @@ def suggest_baseline(
                 specific explainability method. Currently, only SHAP is supported.
             model_config (:class:`~sagemaker.clarify.ModelConfig`): Config of the model and its
                 endpoint to be created.
-            model_scores (int or str): Index or JSONPath location in the model output for the
-                predicted scores to be explained. This is not required if the model output is
-                a single score.
+            model_scores (int or str or :class:`~sagemaker.clarify.ModelPredictedLabelConfig`):
+                Index or JSONPath to locate the predicted scores in the model output. This is not
+                required if the model output isa single score. Alternatively, it can be an instance
```
Comment (navinsoni) [inline src/sagemaker/model_monitor/clarify_model_monitoring.py:838]: Please give space between "is a"

File: src/sagemaker/clarify.py (hunk 1 above)
Comment (xgchena) [reply src/sagemaker/clarify.py]: Good catch. The problem was introduced during rebase/resolving conflicts. The commit is not supposed to change explainability_config parameter so I have restore the dev branch docstring string for the parameter.

File: src/sagemaker/clarify.py (hunk 2 above)
Comment (xgchena) [reply src/sagemaker/clarify.py]: Thanks. Fixed in the new revision.

File: src/sagemaker/model_monitor/clarify_model_monitoring.py (hunk 3 above)
Comment (xgchena) [reply src/sagemaker/model_monitor/clarify_model_monitoring.py]: Thanks. Fixed in the new revision.

**Review by navinsoni — APPROVED** [review]

**Review by BasilBeirouti — APPROVED** [review]

**Review by xiaoyi-cheng — APPROVED** [review]

**Pull request merged**